### ISSUE TITLE
Questions on the CLI

### ISSUE THREAD

**Issue by kalinchernev** [acheong08/ChatGPT#212]
Hi, thank you very much for your work on this project!

When I'm using the CLI, the first Enter goes into a new line and the second Enter sends the message to the chatbot.
However, I can't find a way how to go a line back and edit my previous line before pressing the first Enter. Maybe it's possible but I can't find the way how to do it.

Another question: is it possible somehow to input/output the conversation as in bash? For example < an already created file or > the reply to a file?

**Comment by acheong08** [comment]
> However, I can't find a way how to go a line back and edit my previous line before pressing the first Enter. Maybe it's possible but I can't find the way how to do it.

This is a flaw with the input system of Python. I have limited capacity to fix this. If you know of any good TUI libraries, please inform me.

> Another question: is it possible somehow to input/output the conversation as in bash? For example < an already created file or > the reply to a file?

This is possible if you create a custom client using this library as a dependency. You can check out the developer docs here: https://github.com/acheong08/ChatGPT/wiki/Developer-Docs

**Comment by talwrii** [comment]
> If you know of any good TUI libraries, please inform me.

curses is  the old way of doing. On the other hand if you are starting to implement a bunch of TUI stuff it might be better to integrate with with emacs or vim.


> This is possible if you create a custom client using this library as a dependency.

I created one of those for myself.

https://pypi.org/project/large-language-model-cli/0.10/

This target more CLI that TUI usage and supports this feature as of 30 minutes ago. 

It's worth noting that the concept of a conversation adds little complexity - the client needs to keep track of the converastion between invocations to get the same behaviour as the TUI. I've added this - but there is a bit of a hack where we look at out parents pid to try and use a conversation in a terminal.
